lib/elixir/lib/calendar.ex

```diff
@@ -796,6 +710,75 @@ defmodule NaiveDateTime do
   end
 
   @doc """
+  Adds a specified amount of time to a `NaiveDateTime`.
+
+  Accepts an `integer` in any `unit` available from `System.time_unit`.
+  Negative values will be move backwards in time.
+
+  ## Examples
+
+      # adds seconds by default
+      iex> NaiveDateTime.add(~N[2014-10-02 00:29:10], 2)
+      ~N[2014-10-02 00:29:12]
+      # accepts negative offsets
+      iex> NaiveDateTime.add(~N[2014-10-02 00:29:10], -2)
+      ~N[2014-10-02 00:29:08]
+      # can work with other units
+      iex> NaiveDateTime.add(~N[2014-10-02 00:29:10], 2_000, :milliseconds)
+      ~N[2014-10-02 00:29:12]
+      # doesn't lose precision
```

Comment:
We should likely add tests showing what happens with we use lower precision and higher precision than the one in the NaiveDateTime.